api/scanner/media_encoding/executable_worker/ffmpeg_cli.go

```diff
@@ -0,0 +1,95 @@
+package executable_worker
+
+import (
+	"fmt"
+	"log"
+	"os/exec"
+	"strings"
+
+	"github.com/photoview/photoview/api/utils"
+	"gopkg.in/vansante/go-ffprobe.v2"
+)
+
+type FfmpegCli struct {
+	path       string
+	videoCodec string
+}
+
+func newFfmpegCli() *FfmpegCli {
+	if utils.EnvDisableVideoEncoding.GetBool() {
+		log.Printf("Executable worker disabled (%s=%q): ffmpeg\n", utils.EnvDisableVideoEncoding.GetName(), utils.EnvDisableVideoEncoding.GetValue())
+		return nil
+	}
+
+	path, err := exec.LookPath("ffmpeg")
+	if err != nil {
+		log.Println("Executable worker not found: ffmpeg")
+		return nil
+	}
+
+	version, err := exec.Command(path, "-version").Output()
+	if err != nil {
+		log.Printf("Error getting version of ffmpeg: %s\n", err)
+		return nil
+	}
+
+	codec := utils.EnvVideoCodec.GetValue()
+	fmt.Println("codec:", codec)
+	if codec == "" {
+		codec = "h264"
+	}
+
+	log.Printf("Found executable worker: ffmpeg (%s) with codec %q\n", strings.Split(string(version), "\n")[0], codec)
+
+	return &FfmpegCli{
+		path:       path,
+		videoCodec: codec,
+	}
+}
+
+func (worker *FfmpegCli) IsInstalled() bool {
+	return worker != nil
+}
+
+func (worker *FfmpegCli) EncodeMp4(inputPath string, outputPath string) error {
+	args := []string{
+		"-i",
+		inputPath,
+		"-vcodec", worker.videoCodec,
+		"-acodec", "aac",
+		"-vf", "scale='min(1080,iw)':'min(1080,ih)':force_original_aspect_ratio=decrease:force_divisible_by=2",
```

Comment:
`"string1" + "string2" + "string3"`?

If not limiting with compiling time, `strings.Join(["string1", "string2"], ":")` is also a good way.

Reply:
Actually, this string (and the one for thumbnails) is not so long, so this is an opinionized and optional topic)
But, do you see a value in splitting the string to something like:
```go
"-vf", "scale='min(1080,iw)':'min(1080,ih)'"
    + ":force_original_aspect_ratio=decrease"
    + ":force_divisible_by=2",
```

Reply:
If so, I prefer:

```
        args := []string{
                "-ss", thumbnailOffsetSeconds, // grab frame at time offset
                "-i",
                inputPath,
                "-vframes", "1", // output one frame
                "-an", // disable audio
                "-vf", strings.Join([]string{
                        "scale='min(1024,iw)'",
                        "'min(1024,ih)'",
                        "force_original_aspect_ratio=decrease",
                        "force_divisible_by=2",
                }, ":"),
                outputPath,
        }
```

It won't decrease too much performance.

Reply:
Sure, this is fine too. Just 1 thing: I'd put the `"'min(1024,ih)'",` on the previous line so that each parameter is on its line. It still could be a separate string, but to me, it looks more readable)

Reply:
It's interesting. How did FFMpeg parse `"scale='min(1080,iw)':'min(1080,ih)':force_original_aspect_ratio=decrease:force_divisible_by=2"`? I thought it seperates by `:` in the beginning, but as you said, it could separate like `scale` equals to `'min(1080,iw)':'min(1080,ih)':force_original_aspect_ratio=decrease:force_divisible_by=2`.

If so, I think it doesn't make sense to split lines.

Reply:
According to https://trac.ffmpeg.org/wiki/Scaling, it parses as `scale` equals to `'min(1080,iw)':'min(1080,ih)':force_original_aspect_ratio=decrease:force_divisible_by=2`.

Then splitting the line looks like `force_original_aspect_ratio=decrease` and `force_divisible_by=2` are not `scale` parameters. I hope to keep it in one line.